adapters/repos/db/vector/hnsw/index.go

```diff
@@ -1014,8 +1014,30 @@ func (h *hnsw) Stats() (common.IndexStats, error) {
 }
 
 func (h *hnsw) VectorStorageSize() int64 {
-	// TODO-usage: Implement this
-	return 0
+	h.RLock()
+	defer h.RUnlock()
+
+	// Get count and dimensions
+	var count int64
+	if h.compressed.Load() {
+		count = h.compressor.CountVectors()
```

Comment:
It is not safe to use `CountVectors()` from the cache to count the total vectors as people can set a cache size < total number of buckets. I'd prefer if we instead use the existing `dimensions` bucket (existing approach in shard_dimensions_tracking) to get the total number of vectors and dimensions and purely used the vector indices for compression ratio.